### PR TITLE
Fix small bugs in the class BostedChristyEMPXSec

### PR DESCRIPTION
**Pull request by igor144**
1. Calculate the cross-section in more simple way
2. Fix some problems with uninitialized variables
3. Fix problem with indexing conversion from Fortran into C++
4. Fix separation energy for nuclei with A>55

### CODE REVIEW

File: src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx [hunk 1]
```diff
@@ -164,7 +164,8 @@ double BostedChristyEMPXSec::sigmaNR(int sf, double Q2, double W, bool isDeuteri
   
   double Wdif = W - (Mp + Mpi);
   
-  double m0 = 4.2802; // GeV
+  double m0 = sf==0?0.125:4.2802;
```
Comment (nusense) [inline src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx:167]: Could you make this a bit easier to read by  adding some parentheses and spaces ala 
`double m0 = (sf==0) ?  0.125 : 4.2802;`
(also for the line below for `Q20`)
Also any reference for where these values come from

File: src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx [hunk 2]
```diff
@@ -236,11 +237,14 @@ void BostedChristyEMPXSec::FermiSmearingA(double Q2, double W, double pF, double
   double dW2dpF = 2.*qv;
   double dW2dEs = 2.*(nu + MN); 
   // switched to using 99 bins!
-
-  for (int i=0; i<100; i++)
+  F1p = 0;
+  F1d = 0;
+  sigmaT = 0;
+  sigmaL = 0;
+  for (int i=0; i<99; i++)
```
Comment (nusense) [inline src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx:244]: These seem to make sense, and definitely fix the previous version's overrun of the array.

File: src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx [hunk 3]
```diff
@@ -510,39 +514,35 @@ double BostedChristyEMPXSec::XSec(
...
   double Eprime = E - nu;
   double sin2theta_2 = Q2/4/E/Eprime;
   double cos2theta_2 = 1 - sin2theta_2;
-  double tan2theta_2 = sin2theta_2/cos2theta_2;
-  double eps = 1./(1. + 2*(1+nu2/Q2)*tan2theta_2);              // Ref.1, Eq. (4)
-  double Gamma = kAem*Eprime*K/Q2/E/(1-eps)/2/kPi2;             // Ref.1, Eq. (5)
-  sigmaT = 4*kPi2*kAem*F1/K/MN;
-  sigmaL = R*sigmaT;
-  double xsec = Gamma*(sigmaT+eps*sigmaL);                      // Ref.1, Eq. (3) d2xsec/dOmegadEprime
+  double W2 = W1*(1 + R)/ (1+nu2/Q2);
```
Comment (nusense) [inline src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx:542]: I can't comment on these changes but I'll assume that you've verified them

**Review by nusense — APPROVED** [review]

File: src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx (hunk 1 above)
Comment (igor144) [reply src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx]: Fixed

File: src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx (hunk 3 above)
Comment (igor144) [reply src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx:542]: Yes, I have verified them. I just rewrote the formulas in more compact form

**Review by igor144 — COMMENTED** [review]
Requested chanes.

**Pull request merged**